src/lib/MapStore.ts

```diff
@@ -404,62 +396,58 @@ const mapStore = createStore<MapStore>((set, get) => ({
         missionStore.updateStageArea(missionId, vehicle, stageId, geoCoordinateStructs);
       });
     }
-  },
-  removeStageLayer: () => {},
-  getStageLayer: (missionId: number | null, vehicle: VehicleEnum, stageId: number) => {
+  };
+  const getStageLayer = (missionId: number | null, vehicle: VehicleEnum, stageId: number) => {
     if (missionId === null) return undefined;
-    const mission = get().layerTracking.missions[missionId];
+    const mission = mapState.value.layerTracking.missions[missionId];
     if (!mission) return undefined;
 
     const vehicleLayers = mission.vehicles[vehicle];
     if (!vehicleLayers) return undefined;
 
     return vehicleLayers.stages[stageId];
-  },
-  getZoneLayers: (missionId: number | null, type: ZoneType) => {
+  };
+  const getZoneLayers = (missionId: number | null, type: ZoneType) => {
     if (missionId === null) return [];
-    const mission = get().layerTracking.missions[missionId];
+    const mission = mapState.value.layerTracking.missions[missionId];
     if (!mission) return [];
     return mission.zones[type] as ZoneLayer[];
-  },
-  updateVehicleMarker: (vehicle: VehicleEnum, lat: number, lng: number) => {
-    set((state) => ({
-      vehicleMarkers: {
-        ...state.vehicleMarkers,
-        [vehicle]: L.latLng(lat, lng)
-      }
-    }));
-  },
-  updateMarkerCoords: (vehicle: VehicleEnum, coords: LatLngExpression) => {
+  };
+  const updateVehicleMarker = (vehicle: VehicleEnum, lat: number, lng: number) => {
+    mapState.value.vehicleMarkers = {
+      ...mapState.value.vehicleMarkers,
+      [vehicle]: L.latLng(lat, lng)
+    };
+  };
+  const updateMarkerCoords = (vehicle: VehicleEnum, coords: LatLngExpression) => {
     if (Array.isArray(coords) && coords.length === 2) {
-      set((state) => ({
-        vehicleMarkers: {
-          ...state.vehicleMarkers,
-          [vehicle]: L.latLng(coords[0], coords[1])
-        }
-      }));
+      mapState.value.vehicleMarkers = {
+        ...mapState.value.vehicleMarkers,
+        [vehicle]: L.latLng(coords[0], coords[1])
+      };
     }
-  },
-  getVehicleMarkers: () => {
-    return get().vehicleMarkers;
-  }
-}));
-
-// =============================================
-// Store Export and Watchers
-// =============================================
-const store: MapStore = reactive(mapStore.getState()) as MapStore;
-
-mapStore.subscribe((state) => {
-  Object.assign(store, state);
+  };
+  const getVehicleMarkers = () => {
+    return mapState.value.vehicleMarkers;
+  };
+  const removeStageLayer = () => {} //not sure what is for but it was in the original store
```

Comment:
This is a bit of legacy code that you can remove. removeStageLayer used to update the mapStore state, but pinia now automatically syncs the mapstore to the missionStore.